ui/core/src/api/EthereumService/EthereumService.ts

```diff
@@ -129,82 +127,47 @@ export class EthereumService implements IWalletService {
   }
 
   async connect() {
-    const provider = await this.providerPromise;
     try {
-      if (!provider)
-        throw new Error("Cannot connect because provider is not yet loaded!");
-
-      this.web3 = new Web3(provider);
-
-      // Let's test for Metamask
-      if (isMetaMaskProvider(provider)) {
-        if (provider.request) {
-          // If metamask lets try and connect
-          await provider.request({ method: "eth_requestAccounts" });
-        }
+      if (!this.web3 || !this.provider) {
+        throw new Error("There is no yet a connection to Ethereum.");
+      }
+      if (isMetaMaskProvider(this.provider) && this.provider.request) {
+        await this.provider.request({ method: "eth_requestAccounts" });
       }
-
-      this.addWeb3Subscription();
       notify({ type: "success", message: "Connected to Metamask" });
       await this.updateData();
     } catch (err) {
       console.log(err);
-      this.web3 = null;
+      this.state = reactive({ ...initState });
     }
   }
 
-  addWeb3Subscription() {
-    // TODO: Work out why we cannot subscribe to events when using metamask
-    //       provider and reinstate the following
-    //       Commenting out for now
-    //
-    // this.blockSubscription = this.web3?.eth.subscribe("newBlockHeaders");
-    // this.blockSubscription.on("data", (result: any) => {
-    //   console.log("Ethereum new block header");
-    //   this.updateData();
-    //   this.state.log = result?.hash ?? "null";
-    // });
-
-    // So the above is not working for Metamask instead we setup an interval and do a simple poll
-    let count = 0;
-    let interval = setInterval(() => {
-      this.updateData();
-      this.state.log = `${++count}`;
-    }, 2000);
-    this.blockSubscription = {
-      unsubscribe() {
-        clearInterval(interval);
-      },
-    };
-  }
-
-  removeWeb3Subscription() {
-    this.blockSubscription?.unsubscribe();
-  }
-
   async disconnect() {
     if (isMetaMaskProvider(this.provider)) {
-      this.provider.disconnect &&
-        this.provider.disconnect(0, "Website disconnected wallet");
+      if (this.provider.request) {
+        await this.provider.request({
```

Comment:
This probably needs to be reverted. I found this in the docs too but it doesn't work and asks metamask to reconnect instead of disconnect.

Reply:
@ryardley yes, there is no way to disconnect. But this approach pops up metamask forcing address selection, so you can actually connect a different address. Otherwise, the disconnect button makes it look like you disconnect, but the wallet is actually keeps being connected...